### PR TITLE
Fix routing docs

### PR DESCRIPTION
**Pull request by stevapple**
Fix routing docs and add `getCatchall()` from vapor/routing-kit#94

### CODE REVIEW

File: 4.0/docs/routing.md [hunk 1]
```diff
@@ -196,6 +196,8 @@ app.get("foo", "**") { req in
 
 ### Parameters
 
+#### Parameter
```
Comment (tanner0101) [inline 4.0/docs/routing.md:199]: `Parameters -> Parameter` is kind of strange. I think we could just drop this heading.

File: 4.0/docs/routing.md [hunk 2]
```diff
@@ -22,13 +22,13 @@ http://vapor.codes/hello/vapor
...
 The first part of the request is the HTTP method. `GET` is the most common HTTP method, but there are several you will use often. These HTTP methods are often associated with [CRUD](https://en.wikipedia.org/wiki/Create,_read,_update_and_delete) semantics.
 
-|method|crud|
-|-|-|
-|`GET`|Read|
-|`POST`|Create|
-|`PUT`|Replace|
-|`PATCH`|Update|
-|`DELETE`|Delete|
+| Method   | CRUD    |
```
Comment (tanner0101) [inline 4.0/docs/routing.md:25]: I'd prefer to keep the tables as is, see reasoning here: https://github.com/vapor/docs/pull/500

**Comment by stevapple** [comment]
@tanner0101 Would you please review the changes?

**Comment by stevapple** [comment]
Hello? @tanner0101 Is this fine for merge?

**Pull request merged**